src/Event/EventManager.php

```diff
@@ -361,6 +369,7 @@ public function dispatch($event)
                 $event->result = $result;
             }
         }
+        $this->_dispatchedEvents[] = $event;
```

Comment:
This could end up consuming a non-trivial amount of memory. This needs to be an opt-in feature.